source/system/string/string_literal.c

```diff
@@ -1,15 +1,16 @@
 #include "string_literal.h"
 #include <assert.h>
+#include <stdbool.h>
 #include <ooc/system/String-fwd.h>
 #include <ooc/base/Mutex.h>
 
 #if defined(__WIN32__) || defined(__WIN64__)
 
 typedef HANDLE mutex_t;
-#define MUTEX_INIT NULL
+static HANDLE mutex_init() { return CreateMutex(NULL, FALSE, NULL); }
 static void mutex_lock(mutex_t* m) {
 	if (*m == NULL) {
-		HANDLE mutex = CreateMutex(NULL, FALSE, NULL);
+		HANDLE mutex = mutex_init();
```

Comment:
I think it should be something like
```
static void mutex_init(mutex_t* m) {
    *m = CreateMutex(NULL, FALSE, NULL);
}
```

because of how it's used in line 41